Object Oriented Interface

perlpunk opened the pull request:
This adds an OOP interface and implements the [YAML 1.2 core schema](https://perlpunk.github.io/yaml-test-schema/schemas.html)

perlpunk committed:
Add Object Oriented Interface

perlpunk committed:
Documentation

perlpunk committed:
Add option header

perlpunk committed:
Add option footer

perlpunk committed:
Add option require_footer

perlpunk committed:
Move code to oo_dump_stream

perlpunk committed:
Add option anchor_prefix

perlpunk committed:
Improve error handling

perlpunk committed:
Add option width

perlpunk committed:
v0.904.0

ingydotnet commented on lib/YAML/XS.pod:
```diff
@@ -118,6 +126,157 @@ Sets the number of spaces for indentation for C<Dump>.
+
+=back
+
+=head3 load_string
```
I would prefer this to be `load`. Maybe both is ok.

ingydotnet commented on lib/YAML/XS.pod:
```diff
@@ -118,6 +126,157 @@ Sets the number of spaces for indentation for C<Dump>.
+In scalar context, if the YAML string contains more than one document, it will
+return the first document.
+
+=head3 dump_string
```
Prefer `dump`

perlpunk committed:
s/load_string/load/; s/dump_string/dump/

perlpunk committed:
Fix some C warnings and errors

* Fix grok_* function calls
* Fix loop initialization
* Remove unused vars

perlpunk committed:
Test bools

perlpunk committed:
Fix sporadic segfault

There was a race condition between the exception handler and DESTROY

perlpunk committed:
Use C instead of perl regex to match numbers

perlpunk committed:
Schema: Test more errors

perlpunk committed:
tests cleanup

ingydotnet reviewed (approved):
latm :)

perlpunk commented:
Added me to authors as discussed. Cleaned up a little bit and marked the new interface as experimental for now.
Segfault seems to be gone according to the new cpantesters 💚 results.
https://metacpan.org/release/TINITA/YAML-LibYAML-v0.903.4-TRIAL/view/lib/YAML/XS.pm#OBJECT-ORIENTED-INTERFACE

perlpunk commented:
https://metacpan.org/release/TINITA/YAML-LibYAML-v0.904.0
